File: backend/src/neurocampus/app/schemas/prediccion.py
# backend/src/neurocampus/app/schemas/prediccion.py
from pydantic import BaseModel, Field
from typing import Dict, Any, List, Optional

class OnlineInput(BaseModel):
    calificaciones: Dict[str, float] = Field(default_factory=dict)
    comentario: str = ""

class PrediccionOnlineRequest(BaseModel):
    job_id: Optional[str] = None
    family: str = "sentiment_desempeno"
    input: OnlineInput

class PrediccionOnlineResponse(BaseModel):
    label_top: str
    scores: Dict[str, float]
    sentiment: Dict[str, float]
    confidence: float
    latency_ms: int
    correlation_id: str

class PrediccionBatchItem(BaseModel):
    id: Optional[str] = None
    calificaciones: Dict[str, float] = Field(default_factory=dict)
    comentario: str = ""

class PrediccionBatchResponse(BaseModel):
    batch_id: str
    summary: Dict[str, Any]
    sample: List[Dict[str, Any]]
    artifact: str
    correlation_id: str

File: frontend/src/services/prediccion.ts
// frontend/src/services/prediccion.ts
// Endpoints reales del router /prediccion

import api from "./apiClient";

export type OnlineInput = {
  calificaciones: Record<string, number>;
  comentario: string;
};

export type PrediccionOnlineRequest = {
  job_id?: string | null;
  family?: string; // default backend: "sentiment_desempeno"
  input: OnlineInput;
};

export type PrediccionOnlineResponse = {
  label_top: string;
  scores: Record<string, number>;
  sentiment?: string;
  confidence?: number;
  decision_rule?: Record<string, number> | null;
  latency_ms: number;
  correlation_id: string;
};

export type PrediccionBatchResponse = {
  batch_id: string;
  summary: Record<string, any>;
  sample: Array<Record<string, any>>;
  artifact: string; // URL o path devuelto por backend
  correlation_id: string;
};

export async function online(req: PrediccionOnlineRequest) {
  const { data } = await api.post<PrediccionOnlineResponse>("/prediccion/online", req);
  return data;
}

/**
 * Batch: el backend actual recibe multipart file.
 * Si en el futuro se soporta dataset_id por query, queda listo:
 *   POST /prediccion/batch?dataset_id=...
 */
export async function batch(file: File, opts?: { dataset_id?: string }) {
  const fd = new FormData();
  fd.append("file", file);

  const qs = opts?.dataset_id ? `?dataset_id=${encodeURIComponent(opts.dataset_id)}` : "";
  const { data } = await api.post<PrediccionBatchResponse>(`/prediccion/batch${qs}`, fd);

  return data;
}
